src/gateway/sharding/shard_manager.rs

```diff
@@ -118,6 +118,15 @@ impl ShardManager {
         }
     }
 
+    /// Retrieves a function which can be used to shut down the ShardManager later.
+    ///
+    /// This function will return `true` if the ShardManager has successfully been
+    /// notified to shut down, or false if it has already shut down and been dropped.
+    pub fn get_shutdown_trigger(&self) -> impl FnOnce() -> bool + Send + use<> {
+        let manager_tx = self.manager_tx.clone();
+        move || manager_tx.unbounded_send(ShardManagerMessage::Quit(Ok(()))).is_ok()
+    }
+
```

Comment:
Do you think exposing `self.manager_tx` is maybe a more flexible solution? This function kinda sticks out, and I don't really like the name.

Reply:
It would be more flexible, but it's exposing too much API to public consumers imo, leading to defensive or defective programming inside the currently quite simple `ShardManager`. This is the bare minimum interface that users (at least I) need, and future refactors/reworks cannot be merged if they cannot replicate this behavior.